python/lsst/afw/table/catalogMatches.py

```diff
@@ -254,3 +256,112 @@ def copyAliasMapWithPrefix(inSchema, outSchema, prefix=""):
         outSchema.getAliasMap().set(prefix + k, prefix + v)
 
     return outSchema
+
+
+def reindexCatalog(catalog, indices, deep=True):
+    """Apply a numpy index array to an afw Catalog
+
+    Parameters
+    ----------
+    catalog : subclass of `lsst.afw.table.BaseCatalog`
+        Catalog to reindex.
+    indices : `numpy.ndarray` of `int`
+        Index array.
+
+    Returns
+    -------
+    new : subclass of `lsst.afw.table.BaseCatalog`
+        Reindexed catalog. Records are shallow copies of those in ``catalog``.
+    """
+    new = type(catalog)(catalog.schema)
+    new.reserve(len(indices))
+    for ii in indices:
+        _rec = catalog[int(ii)]
+        new.extend([_rec], deep=deep)
+    return new
+
+
+def matchCatalogsExact(catalog1, catalog2, patch1=None, patch2=None):
+    """Match two catalogs derived from the same mergeDet catalog
+
+    When testing downstream features, like deblending methods/parameters
+    and measurement algorithms/parameters, it is useful to to compare
+    the same sources in two catalogs. In most cases this must be done
+    by matching on either RA/DEC or XY positions, which occassionally
+    will mismatch one source with another.
+
+    For a more robust solution, as long as the downstream catalog is
+    derived from the same mergeDet catalog, exact source matching
+    can be done via the unique ``(parent, deblend_peakID)``
+    combination. So this function performs this exact matching for
+    all sources both catalogs.
+
+    Parameters
+    ----------
+    catalog1, catalog2: subclass of `lsst.afw.table.BaseCatalog`
+        The two catalogs to merge
```

Comment:
I haven't seen this style in numpydoc parameters before.  Are you sure it's allowed?

Also, these aren't actually subclasses; they're instances (subclasses would be types).  But (see below), I think this should just be `SourceCatalog` anyway, because I'm skeptical the return type is compatible with any other catalog type.

Reply:
Good point on the subclasses, I'll update this.

What do you mean about the style for the numpydoc parameters? You mean listing two parameters on the same line? From https://numpydoc.readthedocs.io/en/latest/format.html \:

> When two or more input parameters have exactly the same type, shape and description, they can be combined:
```
x1, x2 : array_like
    Input arrays, description of `x1`, `x2`.
```

Reply:
Note that our docs are not exactly numpydoc format as I found out, so you need to consult our style guide